docs/reference/docs/reindex.asciidoc

```diff
@@ -601,3 +601,29 @@ and it'll look like:
 // TESTRESPONSE
 
 Or you can search by `tag` or whatever you want.
+
+
+[float]
+=== Reindex daily indices
+
+You can use `_reindex` in combination with painless to reindex daily indices to apply a new template.
+Painless extracts the date of the index name and creates a new index with `-1` appended. All data 
+from `metricbeat-2016.05.31` will be reindex into `metricbeat-2016.05.31-1`.
+
```

Comment:
I would try to tell a little story with the snippets. This makes them more self-testing and makes them easier to follow (I hope).

So something like:
```
Say you have indexes that follow a pattern like:
----------------------------------------------------------------
PUT metricbeat-2016.05.30/beat/1?refresh
{"data": "thing1"}
PUT metricbeat-2016.05.31/beat/1?refresh
{"data": "thing2"}
----------------------------------------------------------------
// CONSOLE

You can reindex all the indexes at once with:
[source,js]
----------------------------------------------------------------
POST _reindex?refresh
{
  "source": {
    "index": "metricbeat-*"
  },
  "dest": {
    "index": "metricbeat"
  },
  "script": {
  	"lang": "painless",
    "inline": "ctx._index = 'metricbeat-' + (ctx._index.substring('metricbeat-'.length(), ctx._index.length())) + '-1'"
  }
}
----------------------------------------------------------------
// CONSOLE
// TEST[continued]

Which puts the new documents at:
[source,js]
----------------------------------------------------------------
GET metricbeat-2016.05.30-1/beat/1
GET metricbeat-2016.05.30-2/beat/1
----------------------------------------------------------------
// CONSOLE
// TEST[continued]

As the name implies, this is useful when working with daily indexes like those made by <<link,Beats>> or <<link,Logstash>>. Keep in mind that this executes the reindex on all indexes at once rather than one a time. If this process is going to take a long time you might prefer scripting the wildcard yourself so you get incremental progress.
```